pkg/apis/cr/v1alpha1/types.go

```diff
@@ -134,6 +134,8 @@ type ActionStatus struct {
 
 // ActionProgress provides information on the progress of an action.
 type ActionProgress struct {
+	// OnPhae represents which phase of the action is being run
+	OnPhase string `json:"onPhase,omitempty"`
```

Comment:
Should we name it CurrentPhase instead? Just my personal opinion. OnPhase could be confused with some non-generic field.

Reply:
yeah, I spent sometime trying to come up with a better name but was not able to. I didn't chose `CurrentPhase` because current is in the name and the value is not actually the current phase, value is the phase that was run recently or phase that was run last. I am open to change this btw.
I also considered `runningphase` but that is also a bit confusing.

Reply:
Not a fan of `OnPhase` either. What if we name it `LastCompletedPhase` if we are displaying the recently completed one?

Reply:
We are not actually displaying the recently completed one, we are displaying the currently running phase but we didn't name is `runningPhase` because when actionset completed it pointed to last run phase in the actionset.
I think the best option is to name it `runningPhase` and then change this to `completed` once actionset is completed.

Reply:
I don't think we should change the field to `failed` or `completed`. The `STATE` field already provides that information. How about setting it to `""` after completion or failure?